### ISSUE TITLE
Storage Bucket IAM External Reference Format question

### ISSUE THREAD

**Issue by TravRan** [GoogleCloudPlatform/k8s-config-connector#197]
I'm trying to add a member binding for a Storage Bucket that was created outside of KCC. When I attempt to add the binding I'm getting the error below. In reviewing the Storage Bucket resource it says to specify the name of the bucket for the IAM External Reference Format property. Is there something I'm missing in my YAML? FYI, I've added member binding to a storage bucket that has been created from KCC successfully. 

ERROR

```
error getting policy member: error getting resource config for referenced resource: error retrieving resource 'REDACTED' with gvk 'storage.cnrm.cloud.google.com/v1beta1, Kind=StorageBucket': storagebuckets.storage.cnrm.cloud.google.com "REDACTED" not found
```

Storage Bucket Resource Link
[https://cloud.google.com/config-connector/docs/reference/resources#storagebucket]

YAML
```
apiVersion: iam.cnrm.cloud.google.com/v1beta1
kind: IAMPolicyMember
metadata:
  name: {{iam-policy-member-name}}
spec:
  member: {{sa}}
  role: roles/storage.admin
  resourceRef:
    apiVersion: storage.cnrm.cloud.google.com/v1beta1
    kind: StorageBucket
    name: {{storage-bucket-name}}
```

**Comment by caieo** [comment]
Hi @TravRan, 
If your StorageBucket was created externally, you'll need to indicate that in your YAML. The way to do that is to just replace the line `name: {{storage-bucket-name}}` to `external: {{storage-bucket-name}}`. So your YAML should look something like:

<pre><code>
apiVersion: iam.cnrm.cloud.google.com/v1beta1
kind: IAMPolicyMember
metadata:
  name: {{iam-policy-member-name}}
spec:
  member: {{sa}}
  role: roles/storage.admin
  resourceRef:
    apiVersion: storage.cnrm.cloud.google.com/v1beta1
    kind: StorageBucket
    external: {{storage-bucket-name}}
</code></pre>

**Comment by TravRan** [comment]
I've tried to add the external reference before as well. I've tried using the name of the storage bucket itself and the resourceId described [here.](https://cloud.google.com/storage/docs/json_api/v1/buckets/setIamPolicy). Any idea on the format it's looking for?

error getting policy member: error getting resource config for referenced resource: no ResourceConfig found for kind StorageBucket with an IDTemplate that matches the id {{storage-bucket-name}}.

error getting policy member: error getting resource config for referenced resource: no ResourceConfig found for kind StorageBucket with an IDTemplate that matches the id projects/_/buckets/{{storage-bucket-name}}

**Comment by caieo** [comment]
@TravRan, sorry I lost track of this thread. The format should match whatever is in the 'IAM External Reference Format' row of the (Config Connector StorageBucket resource page)[https://cloud.google.com/config-connector/docs/reference/resource-docs/storage/storagebucket]. 

Could you also provide what version of Config Connector you're on?

**Comment by TravRan** [comment]
@caieo No worries! And I think the version is likely the problem. We are running 1.8.

**Comment by jcanseco** [comment]
Hi @TravRan, I apologize that this fell off our radar. Were you able to resolve your issue?

> And I think the version is likely the problem. We are running 1.8

Yes, there was an issue with `IAMPolicy`/`IAMPolicyMember` resources with external references to `StorageBucket` resources that was fixed in v1.10.

The correct way to write an `IAMPolicyMember` with an external reference to a `StorageBucket` is as caieo outlined in her post. For example:

```yaml
apiVersion: iam.cnrm.cloud.google.com/v1beta1
kind: IAMPolicyMember
metadata:
  name: my-iampolicymember-name
spec:
  member: serviceAccount:my-service-account-name@my-project-id.iam.gserviceaccount.com
  role: roles/storage.admin
  resourceRef:
    apiVersion: storage.cnrm.cloud.google.com/v1beta1
    kind: StorageBucket
    external: my-storage-bucket-name
```

Please let us know if the above works for you.

**Comment by toumorokoshi** [comment]
Closing this issue due to inactivity, and the fact that a fix was provided. Please re-open if you are still having issue with a more recent version of config connector (we're at 1.31.1 as of this comment).
